src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/NetworkAddressChange.Linux.cs

```diff
@@ -44,21 +54,21 @@ public class NetworkChange
                         CreateSocket();
                     }
 
-                    s_addressChangedSubscribers += value;
+                    s_addressChangedSubscribers.TryAdd(value, ExecutionContext.Capture());
```

Comment:
Interesting.  This represents a breaking change from the previous implementation, in that previously if you did:
```C#
NetworkAddressChangedEventHandler handler = ...;
NetworkAddress.NetworkAddressChanged += handler;
NetworkAddress.NetworkAddressChanged += handler;
```
the handler would then be invoked twice, whereas now it'll only be invoked once.  But that "only once" behavior is the same behavior that exists on Windows (both core and netfx), so it makes sense to make this match, and I'm not particularly concerned about this causing any issues, not only because it was already this way on Windows, but because you need to be prepared for any number of these notifications to happen due to the nature of the event.